process/transaction/interceptedTransaction.go

```diff
@@ -15,17 +16,17 @@ import (
 
 // InterceptedTransaction holds and manages a transaction based struct with extended functionality
 type InterceptedTransaction struct {
```

Comment:
do we still need separate struct for each type of transaction ?

Reply:
Unfortunately yes as the validation logic is encoded here. You can see this type of struct as a wrapper over a data-carrier (data.transaction.Transaction) struct that can be used by the processors.